monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/RepositoryZipController.java

```diff
@@ -78,28 +80,58 @@ public void downloadRepository(@PathVariable("tenant") final String tenant, Http
         FileHelper.safeClose(zipOutput);
     }
 
-    @RequestMapping(method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+    @RequestMapping(method = RequestMethod.POST, produces = MediaType.TEXT_PLAIN_VALUE)
     @ResponseBody
     @RolesAllowed(SecurityRoles.ADMIN)
-    public Map<String, String> uploadRepository(@PathVariable("tenant") final String tenant,
+    public String uploadRepository(@PathVariable("tenant") final String tenant,
             @RequestParam("file") final MultipartFile file) throws IOException {
         final TenantContext context = _tenantContextFactory.getContext(tenant);
         final RepositoryFolder rootFolder = context.getTenantRootFolder();
-
+ 
+        if (file.getSize() == 0){
+            return "Failure. The file is empty"; 
+        } 
+        
+        final String contentType = file.getContentType().trim();
+        if ((!contentType.equals("application/zip")) && (!contentType.equals("application/octet-stream"))) {
+            return "Failure. The file isn't a .zip archive";
+        }
+        
+        final File oldRepositoryZipFile = createZipfolder(rootFolder);
```

Comment:
Why do a preemptive copy of the old repository, before first checking if the uploaded zip is a valid repository? Why not only do this if the uploaded zip is a valid repository? In that manner, there is no overhead of creating it and it doesn't have to be deleted in case the uploaded zip is an invalid repository.